.github/workflows/testing.yml

```diff
@@ -74,7 +61,8 @@
        
        - name: spatial versions (if geopandas is installed)
          run: 'python -c "import geopandas; geopandas.show_versions();"'
-         if: contains(matrix.environment-file, 'DEV') || contains(matrix.environment-file, 'BASE')
+         if: contains(matrix.environment-file, 'dev') || contains(matrix.environment-file, 'stable') ||
+             contains(matrix.environment-file, 'oldest')
```

Comment:
Isn't it easier to check if it does not contain no-optional?

Reply:
I remember having a ton of headaches trying to figure this out some time ago. Seems that [`not` is not a recognized option](https://github.com/pysal/spaghetti/actions/runs/7010980374/workflow#L64) within the context, but [is within some contexts](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#not). Maybe I'm missing something super easy?

Reply:
I *FINALLY* worked out the proper syntax after finding [this answer](https://github.com/orgs/community/discussions/25017#discussioncomment-3246240). Why there is no clear example of this in the [GH Actions docs](https://docs.github.com/en/actions/learn-github-actions/expressions) is a mystery to me.